app/src/org/commcare/util/externalizable/AndroidClassHasher.java

```diff
@@ -13,14 +13,13 @@
 
     private static final int CLASS_HASH_SIZE = 4;
 
-    MessageDigest mMessageDigester;
+    private final MessageDigest mMessageDigester;
     
     public AndroidClassHasher() {
         try {
             mMessageDigester = java.security.MessageDigest.getInstance("MD5");
         } catch (NoSuchAlgorithmException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
+            throw new RuntimeException(e);
```

Comment:
sounds right to me (could be useful to add some more context into that error cause `NoSuchAlgorithm` doesn't make sense to those who are new to this codebase)